### ISSUE TITLE
Number schema with unknown format: 'integer' is not valid

### ISSUE THREAD

**Issue by hahyes** [Azure/autorest#4306]
**Before filling a bug**

- [X] have you checked the [faq](https://github.com/Azure/autorest/tree/main/docs/faq.md) for known issues.
- [X] have you checked existing issues

**Describe the bug**

Hi, I'm trying to create client in C# for some REST API linked here: https://developer.allegro.pl/swagger.yaml (Note: this is not my spec)

To do it I'm using command in windows terminal:
```
autorest --csharp --add-credential --input-file=Swagger.yaml --v3 --verbose --debug (added verbose and debug for more information)
```

Sadly it is throwing me error like this:
```
error   | Modeler | Number schema 'schemas:968' with unknown format: 'integer' is not valid
error   | Modeler/UnknownSchemaType | The model schemas:968 does not have a recognized schema type 'number' {"x-ms-metadata":{"apiVersions":["latest"],"filename":["mem:///110?oai3.shaken.json"],"name":"OrderLineItem-quantity","originalLocations":["file:///.../Swagger.yaml#/components/schemas/OrderLineItem-quantity"],"x-ms-secondary-file":false},"type":"number","format":"integer","minimum":0,"description":"quantity","example":1}
...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\modelerfour\dist\main.js - FAILURE  {} Error: Unrecognized schema type:'number' / format: integer {"x-ms-metadata":{"apiVersions":["latest"],"filename":["mem:///110?oai3.shaken.json"],"name":"OrderLineItem-quantity","originalLocations":["file:///.../Swagger.yaml#/components/schemas/OrderLineItem-quantity"],"x-ms-secondary-file":false},"type":"number","format":"integer","minimum":0,"description":"quantity","example":1}
    at ModelerFour.processSchemaImpl (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:1293:11)
    at ProcessingCache.transform (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:152:10)
    at ProcessingCache.process (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:125:62)
    at ModelerFour.processSchema (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:1056:29)
    at ...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:864:28
    at ModelerFour.use (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:307:14)
    at ModelerFour.createObjectSchema (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:863:12)
    at ModelerFour.processObjectSchema (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:916:31)
    at ModelerFour.processSchemaImpl (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:1230:21)
    at ProcessingCache.transform (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:152:10)
PLUGIN FAILURE: Unrecognized schema type:'number' / format: integer {"x-ms-metadata":{"apiVersions":["latest"],"filename":["mem:///110?oai3.shaken.json"],"name":"OrderLineItem-quantity","originalLocations":["file:///.../Swagger.yaml#/components/schemas/OrderLineItem-quantity"],"x-ms-secondary-file":false},"type":"number","format":"integer","minimum":0,"description":"quantity","example":1} , Error: Unrecognized schema type:'number' / format: integer {"x-ms-metadata":{"apiVersions":["latest"],"filename":["mem:///110?oai3.shaken.json"],"name":"OrderLineItem-quantity","originalLocations":["file:///.../Swagger.yaml#/components/schemas/OrderLineItem-quantity"],"x-ms-secondary-file":false},"type":"number","format":"integer","minimum":0,"description":"quantity","example":1}
    at ModelerFour.processSchemaImpl (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:1293:11)
    at ProcessingCache.transform (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:152:10)
    at ProcessingCache.process (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:125:62)
    at ModelerFour.processSchema (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:1056:29)
    at ...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:864:28
    at ModelerFour.use (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:307:14)
    at ModelerFour.createObjectSchema (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:863:12)
    at ModelerFour.processObjectSchema (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:916:31)
    at ModelerFour.processSchemaImpl (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:1230:21)
    at ProcessingCache.transform (...\.autorest\@autorest_modelerfour@4.21.1\node_modules\@autorest\src\modeler\modelerfour.ts:152:10), {}
fatal   | Error: Unrecognized schema type:'number' / format: integer {"x-ms-metadata":{"apiVersions":["latest"],"filename":["mem:///110?oai3.shaken.json"],"name":"OrderLineItem-quantity","originalLocations":["file:///.../Swagger.yaml#/components/schemas/OrderLineItem-quantity"],"x-ms-secondary-file":false},"type":"number","format":"integer","minimum":0,"description":"quantity","example":1}
...\.autorest\@autorest_core@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js - FAILURE Error: Plugin modelerfour reported failure.
    at ...\.autorest\@autorest_core@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:2766:19
    at ScheduleNode (...\.autorest\@autorest_core@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:1316:33)
fatal   | Process() cancelled due to exception : Plugin modelerfour reported failure. / Error: Plugin modelerfour reported failure.
    at ...\.autorest\@autorestcore@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-corets.js:2766:19
    at ScheduleNode (...\.autorest\@autorestcore@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-corets.js:1316:33)
  Error: Plugin modelerfour reported failure.
```

Autorest version (Windows 10):
```
AutoRest code generation utility [cli version: 3.4.0; node: v12.18.3]

Type       Extension Name                           Version      
core       @autorest/core                           3.1.3        
core       @autorest/core                           3.6.3        
extension  @autorest/csharp                         3.0.0-beta.20210925.3
extension  @autorest/modelerfour                    4.21.1       
core       @microsoft.azure/autorest-core           2.0.4417     
extension  @microsoft.azure/autorest.csharp         2.3.82       
extension  @microsoft.azure/autorest.modeler        2.3.55       
```

**Expected behavior**
Generating code properly, I guess.

**Additional context**
I tried few validators of OpenAPI specs and nothing seems to show that problem is in YAML file. I assume there is something wrong on autorest side then.

**Comment by timotheeguerin** [comment]
Hi @hahyes, I have a fix for this, can you follow the instructions here and confirm it is resolved https://github.com/Azure/autorest/pull/4307#issuecomment-927997505

**Comment by hahyes** [comment]
> Hi @hahyes, I have a fix for this, can you follow the instructions here and confirm it is resolved [#4307 (comment)](https://github.com/Azure/autorest/pull/4307#issuecomment-927997505)

@timotheeguerin Hello. I confirm, problem from this issue is resolved in build from #4307. Sadly, another error happened:

```
...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\modelerfour\dist\main.js - FAILURE  {} Error: Not able to process media type plain/text at this moment.
    at ModelerFour.filterMediaTypes (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\modelerfour.ts:1369:21)
    at ModelerFour.processResponses (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\modelerfour.ts:2077:36)
    at ModelerFour.processOperation (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\modelerfour.ts:1677:10)
    at ModelerFour.process (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\modelerfour.ts:2345:14)
    at processRequest (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\plugin-modelerfour.ts:23:31)
    at ...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\libs\extension-base\dist\extension-base.js:55:1
PLUGIN FAILURE: Not able to process media type plain/text at this moment., Error: Not able to process media type plain/text at this moment.
    at ModelerFour.filterMediaTypes (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\modelerfour.ts:1369:21)
    at ModelerFour.processResponses (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\modelerfour.ts:2077:36)
    at ModelerFour.processOperation (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\modelerfour.ts:1677:10)
    at ModelerFour.process (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\modelerfour.ts:2345:14)
    at processRequest (...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\node_modules\@autorest\src\modeler\plugin-modelerfour.ts:23:31)
    at ...\.autorest\@autorest_modelerfour@4.21.3-ci.341336420\libs\extension-base\dist\extension-base.js:55:1, {}
fatal   | Error: Not able to process media type plain/text at this moment.
...\.autorest\@autorest_core@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js - FAILURE Error: Plugin modelerfour reported failure.
    at ...\.autorest\@autorest_core@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:2766:19
    at ScheduleNode (...\.autorest\@autorest_core@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-core_ts.js:1316:33)
fatal   | Process() cancelled due to exception : Plugin modelerfour reported failure. / Error: Plugin modelerfour reported failure.
    at ...\.autorest\@autorestcore@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-corets.js:2766:19
    at ScheduleNode (...\.autorest\@autorestcore@3.6.3\node_modules\@autorest\core\dist\src_lib_autorest-corets.js:1316:33)
  Error: Plugin modelerfour reported failure.
```

Used command:
```
autorest --csharp --add-credential --input-file=swagger.yaml --version:https://tinyurl.com/yhe6mowv --use:https://tinyurl.com/yzfgerxo --verbose --debug
```
